staging/src/k8s.io/kubectl/pkg/drain/drain.go

```diff
@@ -277,8 +277,9 @@ func (d *Helper) evictPods(pods []corev1.Pod, policyGroupVersion string, getPodF
 				} else if apierrors.IsTooManyRequests(err) {
 					fmt.Fprintf(d.ErrOut, "error when evicting pod %q (will retry after 5s): %v\n", pod.Name, err)
 					time.Sleep(5 * time.Second)
-				} else if apierrors.IsForbidden(err) {
-					// an eviction request in a deleting namespace will throw a forbidden error
+				} else if !pod.ObjectMeta.DeletionTimestamp.IsZero() && apierrors.IsForbidden(err) {
```

Comment:
`DeletionTimestamp` is pointier which means can be nil so I'd consider having check for nil too.

Reply:
@soltysh 
I know what you're saying, but golang is weird here: https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/time.go#L61

DeletionTimestamp.IsZero() is used to test nil here.  I think the method is a little redundant, but seems to be common practice in k8s.

Reply:
:+1: thx for pointing this out.

Reply:
For this bit, there was a comment in the first PR about identifying the cause of the forbidden: should we maybe add a
```go
if apierrors.HasStatusCause(err, v1.NamespaceTerminatingCause)
```
?

Reply:
ref: https://github.com/kubernetes/kubernetes/pull/89314#discussion_r395869753

Reply:
Does 'NamespaceTerminatingCause' actually exist anywhere?

Here's all the metav1 causes: https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L883

Reply:
ha yeah, went through that search as well.
`NamespaceTerminatingCause` is defined in https://godoc.org/k8s.io/api/core/v1

Reply:
/hold
to follow up on https://github.com/kubernetes/kubernetes/pull/89314#discussion_r395869753
(at some point the master branch is going to be opened to PRs without milestone 1.20 in them)
/cc @smarterclayton 
but if yall think this is good as is please remove the hold

Reply:
Okay, let me add that...